### PR TITLE
PoC: Vector agent deployment

### PR DESCRIPTION
**Pull request by ffilippopoulos**

### CODE REVIEW

File: vector/agent/namespaced/daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
+apiVersion: apps/v1
```
Comment (george-angel) [inline vector/agent/namespaced/daemonset.yaml:1]: Is there a source for this - or did you compose this yourself?

File: vector/agent/namespaced/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    spec:
+      serviceAccountName: vector
+      initContainers:
+        - name: render-config
+          image: alpine
+          args:
+            - /bin/sh
+            - -c
+            - apk add --no-cache gettext;
+              envsubst < /etc/config-template/agent.yaml > /etc/config-rendered/agent.yaml
```
Comment (george-angel) [inline vector/agent/namespaced/daemonset.yaml:30]: We are not exporting any envvars?

Comment (ffilippopoulos) [reply vector/agent/namespaced/daemonset.yaml:30]: yup, we already discussed that in slack thread. I probably made a syntax mistake when I tried this and fell back to an initContainer and envsubst, but vector should be able to interpolate env vars directly so I will fix

File: vector/agent/namespaced/daemonset.yaml (hunk 1 above)
Comment (ffilippopoulos) [reply vector/agent/namespaced/daemonset.yaml:1]: no, I copied the manifests we deploy already, for example: https://github.com/utilitywarehouse/kubernetes-manifests/blob/master/exp-1-aws/sys-log/vector.yaml and took it from there. I will see if I can find an upstream and wrap it in a make target

Comment (ffilippopoulos) [reply vector/agent/namespaced/daemonset.yaml:1]: https://github.com/vectordotdev/vector/tree/master/distribution/kubernetes there are base manifests here, but from that I strip down a few mount paths we are not using and also change the image to a debian one that comes with journalctl. Given how simple this deployment looks I am not sure this worth following any upstream

File: vector/agent/namespaced/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    metadata:
+      labels:
+        app.kubernetes.io/name: vector
+        app.kubernetes.io/instance: vector
+        app.kubernetes.io/component: Agent
+    spec:
+      serviceAccountName: vector
+      containers:
+        - name: vector
+          image: "timberio/vector:0.39.0-debian"
```
Comment (hectorhuertas) [inline vector/agent/namespaced/daemonset.yaml:25]:
```suggestion
          image: "timberio/vector:0.39.X-debian"
```

File: vector/agent/namespaced/resources/vector.yaml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  kubernetes_logs:
+    type: kubernetes_logs
+
+  systemd_logs:
+    type: journald
+    current_boot_only: true
+    journal_directory: /var/log/journal
+
+transforms:
+  drop_fluentbit:
```
Comment (hectorhuertas) [inline vector/agent/namespaced/resources/vector.yaml:12]: this reads like we are dropping fluentbit logs, while in reality we are dropping any exluded logs (that happen to still use the fluentbit annotation). I'd call this `drop_excluded` instead

File: vector/agent/namespaced/resources/audits.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      .namespace = .kubernetes.pod_namespace
+      .pod_name = .kubernetes.pod_name
+      .container = .kubernetes.container_name
+      .limit_key = string!(.kubernetes.pod_namespace) + "/" + string!(.app)
+      del(.file)
+      del(.kubernetes)
+      del(.source_type)
+      del(.stream)
+
+  throttle_audits:
```
Comment (hectorhuertas) [inline vector/agent/namespaced/resources/audits.yaml:29]: we probably don't want to throttle audits at all?

File: vector/agent/namespaced/resources/audits.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+data_dir: /vector-data-dir
+sources:
+  kubernetes_audit_logs:
+    type: kubernetes_logs
+    extra_namespace_label_selector: "name=sys-falco"
+
+transforms:
+  kubernetes_audits_remap:
```
[the remark this answers is not in the document]
Comment (ffilippopoulos) [reply vector/agent/namespaced/resources/audits.yaml:8]: hmm good q, we probably need to decide on a different set of metadata? It's true that you do not really care which falco pod logged this, but you do care about the hostname (maybe) and ofc the env

File: vector/agent/namespaced/daemonset.yaml (hunk 1 above)
Comment (george-angel) [reply vector/agent/namespaced/daemonset.yaml:1]: Thats fine, I was wondering about labels:

```
    app.kubernetes.io/name: vector
    app.kubernetes.io/instance: vector
    app.kubernetes.io/component: Agent
```

`instance` and `component` have limited value and looked like they came from upstream I didn't think you would add them.

**Review by hectorhuertas — DISMISSED** [review]

**Review by hectorhuertas — APPROVED** [review]

**Review by george-angel — APPROVED** [review]

**Pull request merged**